csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -0,0 +1,954 @@
+                if (depth == GetObjectsDepth.Tables)
+                {
+                    tableColumnsValues.Add(null);
+                }
+                else
+                {
+                    tableColumnsValues.Add(GetColumnSchema(catalog, dbSchema, row["table_name"].ToString(), columnNamePattern));
+                }
+            }
+
+            List<IArrowArray> dataArrays = new List<IArrowArray>
+            {
+                tableNameBuilder.Build(),
+                tableTypeBuilder.Build(),
+                CreateNestedListArray(tableColumnsValues, new StructType(StandardSchemas.ColumnSchema)),
+                CreateNestedListArray(tableConstraintsValues, new StructType(StandardSchemas.ConstraintSchema))
+            };
+
+            return new StructArray(
+                new StructType(StandardSchemas.TableSchema),
+                length,
+                dataArrays,
+                nullBitmapBuffer.Build());
+        }
+
+        private StructArray GetColumnSchema(
+            string catalog,
+            string dbSchema,
+            string table,
+            string columnNamePattern)
+        {
+            StringArray.Builder columnNameBuilder = new StringArray.Builder();
+            Int32Array.Builder ordinalPositionBuilder = new Int32Array.Builder();
+            StringArray.Builder remarksBuilder = new StringArray.Builder();
+            Int16Array.Builder xdbcDataTypeBuilder = new Int16Array.Builder();
+            StringArray.Builder xdbcTypeNameBuilder = new StringArray.Builder();
+            Int32Array.Builder xdbcColumnSizeBuilder = new Int32Array.Builder();
+            Int16Array.Builder xdbcDecimalDigitsBuilder = new Int16Array.Builder();
+            Int16Array.Builder xdbcNumPrecRadixBuilder = new Int16Array.Builder();
+            Int16Array.Builder xdbcNullableBuilder = new Int16Array.Builder();
+            StringArray.Builder xdbcColumnDefBuilder = new StringArray.Builder();
+            Int16Array.Builder xdbcSqlDataTypeBuilder = new Int16Array.Builder();
+            Int16Array.Builder xdbcDatetimeSubBuilder = new Int16Array.Builder();
+            Int32Array.Builder xdbcCharOctetLengthBuilder = new Int32Array.Builder();
+            StringArray.Builder xdbcIsNullableBuilder = new StringArray.Builder();
+            StringArray.Builder xdbcScopeCatalogBuilder = new StringArray.Builder();
+            StringArray.Builder xdbcScopeSchemaBuilder = new StringArray.Builder();
+            StringArray.Builder xdbcScopeTableBuilder = new StringArray.Builder();
+            BooleanArray.Builder xdbcIsAutoincrementBuilder = new BooleanArray.Builder();
+            BooleanArray.Builder xdbcIsGeneratedcolumnBuilder = new BooleanArray.Builder();
+            ArrowBuffer.BitmapBuilder nullBitmapBuffer = new ArrowBuffer.BitmapBuilder();
+            int length = 0;
+
+            string query = string.Format("SELECT * FROM `{0}`.`{1}`.INFORMATION_SCHEMA.COLUMNS WHERE table_name = '{2}'",
```

Comment:
What if "table" contains a quote character? Same concern applies to all of the places where a SQL string is being built. Ideally, we'd be able to leverage a parameter mechanism; the alternative is to carefully confirm the quoting and escaping rules for BigQuery literals and to escape appropriately. (From what I remember, BigQuery bucks the SQL standard of doubling quotes in favor of C-style backslash escapes.)

Reply:
it doesn't look like quotes can be used per https://cloud.google.com/bigquery/docs/tables#table_naming

Reply:
There are still places where the concatenated string come from the calling code, and those will require something to avoid SQL injection.